scripts/sct_testing.py

```diff
@@ -57,37 +57,86 @@ class bcolors:
 # PARSER
 # ==========================================================================================
 def get_parser():
-    # initialize default param
+    import argparse
+
     param_default = Param()
-    # Initialize the parser
-    parser = Parser(__file__)
-    parser.usage.set_description(
-        'Crash and integrity testing for functions of the Spinal Cord Toolbox. Internet connection is required for downloading testing data.')
-    parser.add_option(name="-f",
-                      type_value="str",
-                      description="Test this specific script (do not add extension).",
-                      mandatory=False,
-                      example='sct_propseg')
-    parser.add_option(name="-d",
-                      type_value="multiple_choice",
-                      description="Download testing data.",
-                      mandatory=False,
-                      default_value=param_default.download,
-                      example=['0', '1'])
-    parser.add_option(name="-p",
-                      type_value="folder",
-                      description='Path to testing data. NB: no need to set if using "-d 1"',
-                      mandatory=False,
-                      default_value=param_default.path_data)
-    parser.add_option(name="-r",
-                      type_value="multiple_choice",
-                      description='Remove temporary files.',
-                      mandatory=False,
-                      default_value='1',
-                      example=['0', '1'])
+
+    parser = argparse.ArgumentParser(
+     description="Crash and integrity testing for functions of the Spinal Cord Toolbox. Internet connection is required for downloading testing data.",
+    )
+
+    parser.add_argument("--function", "-f",
+     help="Test this specific script (eg. 'sct_propseg').",
+     nargs="+",
+    )
+
+    def arg_jobs(s):
+        jobs = int(s)
+        if jobs > 0:
+            pass
+        elif jobs == -1:
+            jobs = None
+        else:
+            raise ValueError()
```

Comment:
Missing error message in the `ValueError` exception.

Reply:
It doesn't matter as the message doesn't get brought to the user's attention (argparse is eating it).